treewide: adjust meta descriptions to avoid definite or indefinite articles

philiptaron opened the pull request:
We avoid these articles to conform to [this style guide entry](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#meta-attributes):

    meta.description must:
        Be short, just one sentence.
        Be capitalized.
        Not start with the definite or an indefinite article.
        Not start with the package name.
            More generally, it should not refer to the package name.
        Not end with a period (or any punctuation for that matter).
        Provide factual information.
            Avoid subjective language.


Inspired by a comment by @dotlambda on [`tscli`](https://github.com/NixOS/nixpkgs/pull/413709/files/e32cdad1e8153475ce0157031dc32e32a2cd23e5#r2125239601), which made me think that this would be fairly widespread (as it was.)

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [Nixpkgs 25.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/doc/release-notes/rl-2511.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/doc/manual/release-notes/rl-2505.section.md) Nixpkgs Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
- [NixOS 25.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2511.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) NixOS Release notes)
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

philiptaron commented:
I shouldn't have done the `with lib;` changes. Reverting those.

pbsds commented on pkgs/by-name/db/dbvisualizer/package.nix:
```diff
@@ -64,7 +64,7 @@ stdenv.mkDerivation (finalAttrs: {
 
   meta = {
-    description = "The universal database tool";
+    description = "Universal database tool";
```
i think this changes the description semantically

philiptaron replied:
sadly yes

philiptaron commented on pkgs/by-name/ce/certinfo-go/package.nix:
```diff
@@ -28,7 +28,7 @@ buildGoModule rec {
     changelog = "https://github.com/paepckehh/certinfo/releases/tag/v${version}";
     homepage = "https://paepcke.de/certinfo";
-    description = "Tool to analyze and troubleshoot x.509 & ssh certificates, encoded keys, ...";
+    description = "Tool to analyze and troubleshoot x.509 & ssh certificates, encoded keys, ..";
```
yep, or remove them

dotlambda commented on pkgs/by-name/am/amazon-ecs-cli/package.nix:
```diff
@@ -33,7 +33,7 @@ stdenv.mkDerivation rec {
     homepage = "https://docs.aws.amazon.com/AmazonECS/latest/developerguide/ECS_CLI.html";
     description = "Amazon ECS command line interface";
-    longDescription = "The Amazon Elastic Container Service (Amazon ECS) command line interface (CLI) provides high-level commands to simplify creating, updating, and monitoring clusters and tasks from a local development environment.";
+    longDescription = "Amazon Elastic Container Service (Amazon ECS) command line interface (CLI) provides high-level commands to simplify creating, updating, and monitoring clusters and tasks from a local development environment.";
```
```suggestion
    longDescription = "The Amazon Elastic Container Service (Amazon ECS) command line interface (CLI) provides high-level commands to simplify creating, updating, and monitoring clusters and tasks from a local development environment.";
```
I don't think we have rules about `longDescription`, especially since this one is a full sentence.

dotlambda commented on pkgs/by-name/ck/ckdl/package.nix:
```diff
@@ -59,7 +59,7 @@ pkgs.stdenv.mkDerivation {
 
   meta = {
-    description = "ckdl is a C (C11) library that implements reading and writing the KDL Document Language.";
+    description = "ckdl is a C (C11) library that implements reading and writing the KDL Document Language";
```
```suggestion
    description = "C (C11) library that implements reading and writing the KDL Document Language";
```
> `meta.description` must not start with the package name.

dotlambda commented on pkgs/by-name/cu/culvert/package.nix:
```diff
@@ -36,7 +36,7 @@ stdenv.mkDerivation {
   meta = with lib; {
     homepage = "https://github.com/amboar/culvert";
-    description = "A Test and Debug Tool for BMC AHB Interfaces ";
+    description = "Test and Debug Tool for BMC AHB Interfaces ";
```
```suggestion
    description = "Test and Debug Tool for BMC AHB Interfaces";
```

dotlambda commented on pkgs/by-name/db/dbus-broker/package.nix:
```diff
@@ -39,7 +39,7 @@ let
       inherit buildInputs;
       meta = meta // {
-        description = "The C-Util Project is a collection of utility libraries for the C11 language.";
+        description = "C-Util Project is a collection of utility libraries for the C11 language";
```
```suggestion
        description = "Collection of utility libraries for the C11 language";
```

dotlambda commented on pkgs/by-name/do/dotnet-ef/package.nix:
```diff
@@ -7,7 +7,7 @@ buildDotnetGlobalTool {
 
   meta = {
-    description = "The Entity Framework Core tools help with design-time development tasks.";
+    description = "Entity Framework Core tools help with design-time development tasks";
```
```suggestion
    description = "Tools that help with design-time development tasks";
```

dotlambda reviewed (commented):
I assume the rebuilds are there because `meta.description` is used in desktop files?

pbsds reviewed (approved):
I ![image](https://github.com/user-attachments/assets/7e16d60f-0fa8-437a-9868-fb2d3fd24dd8) the whole patch, other than the two nits this LGTM.

Not sure where the rebuilds come from however...

dotlambda commented on pkgs/by-name/bi/bibutils/package.nix:
```diff
@@ -45,7 +45,7 @@ stdenv.mkDerivation rec {
   meta = with lib; {
     description = "Bibliography format interconversion";
-    longDescription = "The bibutils program set interconverts between various bibliography formats using a common MODS-format XML intermediate. For example, one can convert RIS-format files to Bibtex by doing two transformations: RIS->MODS->Bibtex. By using a common intermediate for N formats, only 2N programs are required and not N²-N. These programs operate on the command line and are styled af ...
+    longDescription = "Bibutils program set interconverts between various bibliography formats using a common MODS-format XML intermediate. For example, one can convert RIS-format files to Bibtex by doing two transformations: RIS->MODS->Bibtex. By using a common intermediate for N formats, only 2N programs are required and not N²-N. These programs operate on the command line and are styled after  ...
```
This does not make sense grammatically.
Let's just not touch `longDescription` at all.

philiptaron committed:
treewide: adjust meta descriptions to avoid definite or indefinite articles

This patch was produced in Vim by me, a human being. All errors are the fault of the operator. I did try to be careful.

Co-authored-by: Peder Bergebakken Sundt <pbsds@hotmail.com>
Co-authored-by: dotlambda <nix@dotlambda.de>

philiptaron commented:
> Not sure where the rebuilds come from however...

It's all `.desktop` entries.

1. I downloaded the [comparison results](https://github.com/NixOS/nixpkgs/actions/runs/15472034456/artifacts/3269422975)
2. On master, I generated all the relevant drvs for those paths: `NIXPKGS_ALLOW_INSECURE=1 NIXPKGS_ALLOW_UNFREE=1 nix-instantiate --strict --json $(jq -r '.attrdiff.changed[]' ~/Downloads/comparison/changed-paths.json | sed 's,^,-A ,')`
3. Did the same on the branch
4. Ran `nix-diff` on the results: `python -c 'b = [s.rstrip() for s in open("/tmp/before.drvs").readlines()]; a = [s.rstrip() for s in open("/tmp/after.drvs").readlines()]; print("\n".join(f"nix-diff {before} {after}" for before, after in zip(b, a)))' | bash`
5. Read through them.

Every single one was a `.desktop` entry. Common, evidently.
